includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -761,6 +761,11 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 			return false;
 		}
 
+		// If this order is not authorized yet, don't try and refund it. Instead, return an appropriate error message.
+		if ( 'requires_capture' === $order->get_meta( '_intention_status', true ) ) {
+			return new WP_Error( 'on-hold-payment', "As this payment is currently on-hold and has not been charged yet, it is not possible to refund this order. To cancel this order, use the 'Cancel Authorization' option in the 'Order Actions' drop-down menu." );
```

Comment:
I wasn't sure on the best wording for this error message, so if anyone has any suggestions they'd be appreciated! :)

Reply:
Good start!

Here are my feedbacks:
- I think 'on hold' is the status of the order, not the payment. If my assumption is correct, then the error code might be better to be `uncaptured-payment`.
- I think it's better to use the term 'authorized' and 'captured' because the option from the settings page is called 'Manual capture', so the user would know what's going on if we use that term. The order note also emphasizes those two terms.
- The user reaches this point because they want to make a refund (could be partial, could be in full amount) and that does not always mean they want to cancel the order. In the case where they want to make partial refund, it might be best to direct them to capture the payment first and then refund. For full refund, suggest to cancel authorization.
- Wrap the string with `__()` for localization. As an example, [you can find from the same file](https://github.com/Automattic/woocommerce-payments/pull/1192/files#diff-bcfd303d507c33a27671935e724ee64d0df30ff2a18bb6ef4c94f28d48190124R792-R793).

I'm not sure what the text copy should be, but if I have to come up with one, I'd use this:
```
This payment is not captured yet. To proceed with a refund, from 'Order Actions', either 'Capture charge' or 'Cancel authorization'.
```

Reply:
Thanks @shendy-a8c for the great feedback.

> I think 'on hold' is the status of the order, not the payment. If my assumption is correct, then the error code might be better to be uncaptured-payment.

Agreed, that makes more sense.

> I think it's better to use the term 'authorized' and 'captured' because the option from the settings page is called 'Manual capture', so the user would know what's going on if we use that term. The order note also emphasizes those two terms.

That sounds good to me to make it more consistent.

> The user reaches this point because they want to make a refund (could be partial, could be in full amount) and that does not always mean they want to cancel the order. In the case where they want to make partial refund, it might be best to direct them to capture the payment first and then refund. For full refund, suggest to cancel authorization.

True. I would assume that most of the time a user would be confused if they are trying to refund an order which is authorized but not yet captured (they may not understand what this means). Perhaps it would be worth also looking at our documentation and making sure that we have some discussion about what this means and when to do a refund vs when to cancel an order?

> Wrap the string with __() for localization. As an example, you can find from the same file.

Thanks! Will do.

Also, your suggested text copy looks good to me. I will implement that for now unless there are any further comments from the team. 👍 

Also need to work out why the E2E tests failed, so I'll take another pass at this on Monday and make those changes. Thanks again!

Reply:
Updated with these changes now. :)